src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -817,9 +831,22 @@ int doFinal(byte[] input, int inputOffset, int inputLen, byte[] output,
         int estOutSize = getOutputSizeByOperation(inputLen, true);
         int outputCapacity = checkOutputCapacity(output, outputOffset,
                 estOutSize);
+        int outLen = 0;
+
+        if (!decrypting && (padding != null) && (inputLen > 0)) {
+            /*
+            * Call the update() method to get rid of as many bytes as
+            * possible before potentially copying array to pad.
+            */
+            int updateOutLen = update(input, inputOffset, inputLen, output, outputOffset);
+            inputOffset = inputLen;
+            inputLen = 0;
+            outputOffset += updateOutLen;
+            outLen = updateOutLen;
```

Comment:
Extra local variable `updateOutLen` is not needed:
```java
            outLen = update(input, inputOffset, inputLen, output, outputOffset);
            outputOffset += outLen;
            inputOffset = inputLen;
            inputLen = 0;
```

Reply:
Same as above.